src/scripts/nginx-install.sh

```diff
@@ -0,0 +1,186 @@
+log()
+{
+    echo \[$(date +%d%m%Y-%H:%M:%S)\] "$1"
+    echo \[$(date +%d%m%Y-%H:%M:%S)\] "$1" >> /var/log/nginx-install.log
+}
+
+#########################
+# Parameter handling
+#########################
+
+AUTH_USERNAME=""
+AUTH_PASSWORD=""
+NGINX_PORT=0
+PFX_B64=""
+
+#Loop through options passed
+while getopts :u:p:n:c:h optname; do
+  log "Option $optname set"
+  case $optname in
+    u) # basic auth user name
+      AUTH_USERNAME="${OPTARG}"
+      ;;
+    p) # basic auth password
+      AUTH_PASSWORD="${OPTARG}"
+      ;;
+    n) # TCP port number
+      NGINX_PORT=${OPTARG}
+      ;;
+    c) # base64 encoded PFX file
+      PFX_B64="${OPTARG}"
+      ;;
+    h) #show help
+      help
+      exit 2
+      ;;
+    \?) #unrecognized option - show help
+      echo -e \\n"ERROR: unknown option -${BOLD}$OPTARG${NORM}" >&2
+      help
+      exit 2
+      ;;
+  esac
+done
+
+
+if [ "${AUTH_USERNAME}" == "" -o "${AUTH_PASSWORD}" == "" -o ${NGINX_PORT} -eq 0 -o "${PFX_B64}" == "" ];
+then
+    echo "ERROR: missing input arguments" >&2
+    help
+    exit 3
+fi
+
+#########################
+# Constants
+#########################
+
+NGINX_DIR=/etc/nginx
+NGINX_LOG_DIR=/var/log/nginx
+BASIC_AUTH_PATH=$NGINX_DIR/.htpasswd
+CERT_DIR=$NGINX_DIR/ssl
+CRT_PATH=$CERT_DIR/host.crt
+KEY_PATH=$CERT_DIR/host.key
+DEFAULT_SITE_PATH=$NGINX_DIR/sites-available/default
+ES_LOCAL_ADDRESS=http://localhost:9200
+
+#########################
+# Installation functions
+#########################
+
+install_nginx()
+{
+  apt-get -yq install nginx
+}
+
+restart_nginx()
+{
+  systemctl restart nginx
+}
+
+create_basic_auth_file()
+{
+  htpasswd -cb $BASIC_AUTH_PATH $AUTH_USERNAME $AUTH_PASSWORD
+}
+
+install_certificate()
+{
+  mkdir $CERT_DIR
+  echo ${PFX_B64} | base64 -d >> $CERT_DIR/host.pfx
+  openssl pkcs12 -in $CERT_DIR/host.pfx -clcerts -nokeys -out $CRT_PATH -passin pass:
+  openssl pkcs12 -in $CERT_DIR/host.pfx -out $KEY_PATH -passin pass: -passout pass:
+  rm $CERT_DIR/host.pfx
+}
+
+write_server_config()
+{
+  rm $NGINX_DIR/nginx.conf
+
+  echo "user www-data;
+worker_processes auto;
+pid /run/nginx.pid;
+
+events {
+	worker_connections 768;
+}
+
+http {
+	sendfile on;
+	tcp_nopush on;
+	tcp_nodelay on;
+	keepalive_timeout 65;
+	types_hash_max_size 2048;
+
+	default_type application/octet-stream;
+
+	gzip on;
+	gzip_disable "msie6";
+
+	ssl_protocols TLSv1.2 TLSv1.3; # omit SSLv3 because of POODLE (CVE-2014-3566)
+	ssl_prefer_server_ciphers on;
+	ssl_certificate $CRT_PATH;
+	ssl_certificate_key $KEY_PATH;
+
+	access_log $NGINX_LOG_DIR/access.log;
+	error_log $NGINX_LOG_DIR/error.log;
+
+	include $NGINX_DIR/conf.d/*.conf;
+	include $NGINX_DIR/sites-enabled/*;
+	include $NGINX_DIR/mime.types;
+}" >> $NGINX_DIR/nginx.conf
+
+}
+
+write_site_config()
+{
+  rm $DEFAULT_SITE_PATH
+
+  echo "server {
+	listen $NGINX_PORT ssl default_server;
+	listen [::]:$NGINX_PORT ssl default_server;
+	location / {
+		proxy_pass $ES_LOCAL_ADDRESS;
+		auth_basic "ElasticSearch!";
```

Comment:
Is this `auth_basic` value always going to be this hardcoded string?

Reply:
This is just a string that shows in a creds prompt. It can be anything, no specific value required.